### PR TITLE
Feature/condo/sberdoma 762/new user register screen

### PR DESCRIPTION
**Pull request by Dimitreee**
Screenshot 2021-07-20 at 17 54 45
Screenshot 2021-07-20 at 17 51 14
Screenshot 2021-07-20 at 17 51 38
Screenshot 2021-07-20 at 17 53 39
Screenshot 2021-07-20 at 17 54 24

### CODE REVIEW

File: apps/condo/pages/billing/trial.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                                        </Space>
+                                    </FocusContainer>
+                                </Col>
+                            )
+                        })}
+                    </Row>
+                </Col>
+                <Col span={24} >
+                    <Button type={'sberPrimary'} onClick={handleSubscribe}>
+                        Начать пробный период
```
Comment (dkoviazin) [inline apps/condo/pages/billing/trial.tsx:100]: Please, move to translates

File: apps/condo/pages/auth/change-password.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ import { useLazyQuery, useMutation } from '@core/next/apollo'
 import { CHANGE_PASSWORD_WITH_TOKEN_MUTATION, CHECK_PASSWORD_RECOVERY_TOKEN } from '@condo/domains/user/gql'
 import { useAuth } from '@core/next/auth'
 import { BasicEmptyListView } from '@condo/domains/common/components/EmptyListView'
+import { AuthLayoutContext } from '../../domains/user/components/containers/AuthLayoutContext'
```
Comment (dkoviazin) [inline apps/condo/pages/auth/change-password.tsx:16]: ../../domains/user/ to @condo/domains/

File: apps/condo/pages/auth/forgot.tsx [hunk 3]
```diff
@@ -15,12 +15,11 @@ import { WRONG_EMAIL_ERROR } from '@condo/domains/user/constants/errors'
 import { getClientSideSenderInfo } from '@condo/domains/common/utils/userid.utils'
 import { LOCK_TIMEOUT } from '@condo/domains/user/constants/common'
 import { CountDownTimer } from '@condo/domains/common/components/CountDownTimer'
-
+import { AuthLayoutContext } from '../../domains/user/components/containers/AuthLayoutContext'
```
Comment (dkoviazin) [inline apps/condo/pages/auth/forgot.tsx:18]: ../../domains/user/ to @condo/domains/

**Review by dkoviazin — APPROVED** [review]
Looks good on almost all top 10 resolutions
top10

1280x720 = Buttons have no space between them

1280x720

360x640 = Footer too high

360x640

**Comment by Dimitreee** [comment]
@dkoviazin  https://doma.atlassian.net/browse/SBERDOMA-824

File: apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        const variables = { data: { token, sender, captcha } }
+        // @ts-ignore TODO(Dimitreee): remove after runMutation typo
+        return runMutation({
+            mutation: resendSmsMutation,
+            variables,
+            intl,
+            form,
+            ErrorToFormFieldMsgMapping,
+        }).catch(error => {
+            console.error(error)
```
Comment (pahaz) [inline apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx:92]: May be add notification to user?

File: apps/condo/domains/user/components/auth/hooks.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            phone: [{ required: true }],
+            name: [
+                {
+                    required: true,
+                    message: PleaseInputYourNameMsg,
+                    whitespace: true,
+                    type: 'string',
+                }, {
+                    message: NameContainsOnlyMsg,
+                    pattern: /^([a-zA-Zа-яА-ЯЁё\s][-'’`]?)+$/,
```
Comment (pahaz) [inline apps/condo/domains/user/components/auth/hooks.ts:34]: @MrFoxPro it’s should depends on locale

File: apps/condo/domains/user/components/containers/AuthLayoutContext.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        // @ts-ignore TODO(Dimitreee): remove after runMutation typo
+        return runMutation({
+            mutation: signinByPhoneMutation,
+            variables: { phone, password },
+            onCompleted: () => {
+                refetch()
+            },
+            intl,
+        }).catch(error => {
+            console.error(error)
```
Comment (pahaz) [inline apps/condo/domains/user/components/containers/AuthLayoutContext.tsx:40]: User notification ?

File: apps/condo/domains/user/components/containers/AuthLayoutContext.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        // @ts-ignore TODO(Dimitreee): remove after runMutation typo
+        return runMutation({
+            mutation: signinByEmailMutation,
+            variables: { email, password },
+            onCompleted: () => {
+                refetch()
+            },
+            intl,
+        }).catch(error => {
+            console.error(error)
```
Comment (pahaz) [inline apps/condo/domains/user/components/containers/AuthLayoutContext.tsx:54]: Too

File: apps/condo/lang/en.json [hunk 8]
```diff
@@ -358,6 +361,7 @@
   "ReadMore": "Expand",
   "Redirecting": "Redirecting ...",
   "Register": "Register",
+  "SberIdRegister": "Регистрация через Сбер ID",
```
Comment (leonid-d) [inline apps/condo/lang/en.json:364]: translate to English?

File: apps/condo/domains/user/components/auth/InputPhoneForm.tsx [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+                        <Col span={24}>
+                            <Form.Item
+                                name='phone'
+                                label={PhoneMsg}
+                                rules={[
+                                    {
+                                        required: true,
+                                        message: FieldIsRequiredMsg,
+                                    },
+                                    () => ({
```
Comment (leonid-d) [inline apps/condo/domains/user/components/auth/InputPhoneForm.tsx:109]: This arrow function seems unnecessary

**Review by AntonAL — CHANGES_REQUESTED** [review]
SberDoma logo has white color in these screens in Figma.

File: apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx (hunk 4 above)
Comment (Dimitreee) [reply apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx:92]: @pahaz i'm not sure i knew how exactly and what about notify user, guess @dkoviazin can help's here.

File: apps/condo/domains/user/components/auth/hooks.ts (hunk 5 above)
Comment (Dimitreee) [reply apps/condo/domains/user/components/auth/hooks.ts:34]: guess @leonid-d can helps here

File: apps/condo/domains/user/components/containers/AuthLayoutContext.tsx (hunk 6 above)
Comment (Dimitreee) [reply apps/condo/domains/user/components/containers/AuthLayoutContext.tsx:40]: @pahaz i'm not sure i knew how exactly and what about notify user, guess @dkoviazin can help's here.

File: apps/condo/domains/user/components/containers/AuthLayoutContext.tsx (hunk 7 above)
Comment (Dimitreee) [reply apps/condo/domains/user/components/containers/AuthLayoutContext.tsx:54]: @pahaz i'm not sure i knew how exactly and what about notify user, guess @dkoviazin can help's here.

File: apps/condo/domains/common/components/PhoneInput.tsx [hunk 10]
```diff
@@ -17,11 +27,25 @@ type PhoneInputRef = {
...
+    let height = '32px'
+
+    if (size === 'large') {
+        height = '40px'
+    } else if (size === 'small') {
+        height = '22px'
+    }
+
+    return { ...style, height }
+}
```
Comment (AntonAL) [inline apps/condo/domains/common/components/PhoneInput.tsx:42]: As for me, this looks simpler:

```suggestion
const getPhoneInputStyles = (style, size: SizeType) => ({
    ...style,
    height: {
        large: '40px',
        small: '22px',
    }[size] || '32px',
})
```

**Review by AntonAL — APPROVED** [review]

**Pull request closed without merging**